File: app/src/main/java/com/codemate/brewflop/injection/NetModule.java
package com.codemate.brewflop.injection;

import com.codemate.brewflop.data.network.SlackApi;
import com.codemate.brewflop.data.network.SlackService;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by iiro on 12.10.2016.
 */
@Module
public class NetModule {
    String baseUrl;

    public NetModule(String baseUrl) {
        this.baseUrl = baseUrl;
    }

    @Provides
    @Singleton
    SlackApi provideSlackApi() {
        return new SlackService(baseUrl).getApi();
    }
}

File: app/src/main/java/com/codemate/brewflop/di/NetModule.kt
package com.codemate.brewflop.di

import com.codemate.brewflop.data.network.SlackService
import dagger.Module
import dagger.Provides
import okhttp3.HttpUrl
import javax.inject.Singleton

@Module
class NetModule(val baseUrl: HttpUrl) {
    @Provides
    @Singleton
    fun provideApi() = SlackService.getApi(baseUrl)
}